frontend/public/components/masthead.tsx

```diff
@@ -9,12 +10,133 @@ import * as dedicatedLogoImg from '../imgs/openshift-dedicated-logo.svg';
+    this.openDocumentation = this.openDocumentation.bind(this);
+  }
+
+  closeAboutModal() {
+    this.setState({ showAboutModal: false });
+  }
+
+  openAboutModal() {
+    this.setState({ showAboutModal: true });
+  }
+
+  openDocumentation() {
+    window.open(openshiftHelpBase, '_blank').opener = null;
+  }
+
+  render() {
+    const {logoAlt, logoImg, kubernetesVersion, openshiftVersion} = this.props;
+    const {showAboutModal} = this.state;
+    return <React.Fragment>
+      <ActionsMenu
+        actions={[
+          {label: 'Documentation', callback: this.openDocumentation},
+          {label: 'About', callback: this.openAboutModal}]}
+        noButton
+        noCaret
+        title={<i className="fa fa-question-circle-o co-masthead__help-icon" />} />
+      <AboutModal className="co-masthead__modal" logo={logoImg} altLogo={logoAlt} productTitle="OpenShift" show={showAboutModal} onHide={this.closeAboutModal}>
+        <h2>About</h2>
+        <p>OpenShift is Red Hat&apos;s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.</p>
+        {(openshiftVersion || kubernetesVersion) &&
+          <React.Fragment>
+            <h2>Version</h2>
+            <AboutModal.Versions className="co-masthead__modal--version">
+              {openshiftVersion && <AboutModal.VersionItem label="OpenShift Master" versionText={openshiftVersion} />}
+              {kubernetesVersion && <AboutModal.VersionItem label="Kubernetes Master" versionText={kubernetesVersion} />}
+            </AboutModal.Versions>
+          </React.Fragment>}
+      </AboutModal>
+    </React.Fragment>;
+  }
+}
+
+const LogoDetails = () => {
+  let logoImg, logoAlt;
+
+  // Webpack won't bundle these images if we don't directly reference them, hence the switch
+  switch ((window as any).SERVER_FLAGS.branding) {
+    case 'ocp':
+      logoImg = ocpLogoImg;
+      logoAlt = 'OpenShift Container Platform';
+      break;
+    case 'online':
+      logoImg = onlineLogoImg;
+      logoAlt = 'OpenShift Online';
+      break;
+    case 'dedicated':
+      logoImg = dedicatedLogoImg;
+      logoAlt = 'OpenShift Dedicated';
+      break;
+    default:
+      logoImg = okdLogoImg;
+      logoAlt = 'OKD';
+  }
+
+  return ({'logoImg': logoImg, 'logoAlt': logoAlt});
+};
+
+const HelpMenuWrapper = connectToFlags(FLAGS.OPENSHIFT)(
+  class _HelpMenuWrapper extends SafetyFirst<FlagsProps, HelpMenuWrapperState> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        openshiftVersion: null,
+        kubernetesVersion: null,
+      };
+    }
+
+    componentDidMount() {
+      super.componentDidMount();
+      this._checkKubernetesVersion();
+      this._checkOpenShiftVersion();
+    }
+
+    componentDidUpdate(prevProps) {
+      if (prevProps.flags[FLAGS.OPENSHIFT] !== this.props.flags[FLAGS.OPENSHIFT]) {
+        this._checkOpenShiftVersion();
+      }
+    }
+
+    _checkOpenShiftVersion() {
```

Comment:
The version logic should be in the modal, not in the menu. We want to check the version when the modal is opened.